Sources/TuistLoader/Loaders/RecursiveManifestLoader.swift

```diff
@@ -68,19 +70,33 @@ public class RecursiveManifestLoader: RecursiveManifestLoading {
 
             let project = try manifestLoader.loadProject(at: path)
             cache[path] = project
-            paths.append(contentsOf: try dependencyPaths(for: project, path: path))
+            paths.append(contentsOf: try dependencyPaths(for: project, path: path, dependenciesGraph: dependenciesGraph))
         }
 
         return LoadedProjects(projects: cache)
     }
 
-    private func dependencyPaths(for project: Project, path: AbsolutePath) throws -> [AbsolutePath] {
+    private func dependencyPaths(
+        for project: ProjectDescription.Project,
+        path: AbsolutePath,
+        dependenciesGraph: DependenciesGraph
+    ) throws -> [AbsolutePath] {
         let generatorPaths = GeneratorPaths(manifestDirectory: path)
         let paths: [AbsolutePath] = try project.targets.flatMap {
             try $0.dependencies.compactMap {
                 switch $0 {
                 case let .project(target: _, path: projectPath):
                     return try generatorPaths.resolve(path: projectPath)
+                case let .external(name):
+                    guard let dependency = dependenciesGraph.externalDependencies[name] else {
+                        return nil
+                    }
+
+                    switch dependency {
+                    // TODO: invoke return try generatorPaths.resolve(path: projectPath) for source based dependencies
+                    case .xcframework:
+                        return nil
+                    }
```

Comment:
If I followed along, is the idea to support something like:

```swift
// Project.swift
// ...

Target(
   dependencies: [
       .external("SourceBasedDependency")
   ]
)
```

Could this mapping also take place in the `ManifestMapper` too?
 
https://github.com/tuist/tuist/pull/3093/files#diff-1779b94775d634799634b9533a39bb42222c4da26a2e6ab8c85147f6d3672be8R60

Where `.external("SourceBasedDependency")` would be replaced with `.project(target: externalTargetName, path: externalTargetPath)` ?

If the goal is to ensure those external manifests are loaded - that could possibly be done in a secondary pass. Would something like this be feasible?

```swift
let manifests = try recursiveManifestLoader.loadProject(at: path)
let externalManifestsPaths = try extractAllSourceBasedExternalDependencies(from: manifests, dependenciesGraph: dependenciesGraph)
let externalManifests = try recursiveManifestLoader.loadProjects(at: externalManifestsPaths)
let allManifests = manifests.merging(externalManifests) 
```

Reply:
Yes, the goal is to load external manifests.
The reason I thought it was more appropriate there, is that an SPM package might define a lot of products, and you might be interested only in a small subset of them.
Putting the logic in the recursive manifest loader makes sure that we only load the manifests we actually need.

Reply:
would the situation change here if `DependenciesGraph` consisted of `TargetDependency` directly? afaict, if we did that, we might be able to leverage more logic for loading of `DependenciesGraph` and `Graph`?

Reply:
Here we have loading from the ProjectDescription, so we still have the `ProjectDescription.TargetDependency.external` entry which have not been mapped to `Graph.TargetDependency.project`.
When we found that, we will need to load its project so that it will be available later on.